### PR TITLE
WIP: Compatibility with Tensorflow 1.0

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

File: src/ops/sequences.jl [hunk 1]
```diff
@@ -230,9 +247,9 @@ Returns:
 @op function Base.reverse(x::AbstractTensor, indices; name=nothing)
     local desc
     with_op_name(name, "Reverse") do
-        desc = NodeDescription("Reverse")
+        desc = NodeDescription("ReverseV2")
         add_input(desc, Tensor(x))
-        add_input(desc, Tensor(indices))
+        add_input(desc, cast(Tensor(indices), Int32)-1)
```
Comment (oxinabox) [inline src/ops/sequences.jl]: Can this pattern `cast(Tensor(indices), Int32)-1)` be extracted out into a function.
It occurs in many places.

File: src/ops/transformations.jl [hunk 2]
```diff
@@ -153,43 +153,90 @@ https://www.tensorflow.org/versions/r0.10/api_docs/python/array_ops.html#split
...
+
+That is, the data from the input tensors is joined along the `axis`
+dimension.
+
+The number of dimensions of the input tensors must match, and all dimensions
+except `axis` must be equal.
+
+For example:
+
+```python
```
Comment (oxinabox) [inline src/ops/transformations.jl]: These examples need translating.
Particularly to 1-based indexing

**Comment by malmaud** [comment]
All good points.

**Comment by malmaud** [comment]
Alright, I'll merge this once CI is green unless there are other comments. I know not everything with the docs and casting is perfect yet, but those can be addressed on `master`.

**Pull request merged**